### PR TITLE
Add support for jobFilters in sectioned view and list view

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: docs/View-Reference.md [hunk 1]
```diff
@@ -18,6 +18,21 @@ view(type: ListView) {  // since 1.21
         names(String... jobNames)
         regex(String regex)
     }
+    jobFilters { // since 1.28
+        regex {
+            includeMatched() | includeUnmatched() | excludeMatched() | excludeUnmatched() // what to do with matching Jobs
+            type(String type) // what to match - one of 'NAME', 'DESCRIPTION', 'SCM', 'EMAIL', 'MAVEN', 'SCHEDULE', 'NODE'
```
Comment (daspilker) [inline docs/View-Reference.md]: This is confusing. The UI calls this "Match Value", but you called it `type` and "Match Type" in the UI is one of the "Include Matched", "Exclude Matched" etc values. How about changing it  to `value` or `matchValue`?

And why is are there explict methods for the "Match Types", but not for the "Match Values"? I would prefer only one method for the "Match Type" as well, so it's more obvious that only one value can be selected.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/views/jobfilter/AbstractJobFilter.groovy [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        new NodeBuilder()."${className}" {
+            addArgs(delegate)
+        }
+    }
+
+    protected void addArgs(NodeBuilder builder) {
+        builder.includeExcludeTypeString(includeExcludeType)
+    }
+
+    Object methodMissing(String name, args) {
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/views/jobfilter/AbstractJobFilter.groovy]: Please use explicit methods to enable IDE support, also in JobStatusesFilter.

**Comment by wolfs** [comment]
Better now?

**Comment by daspilker** [comment]
I rebased and merged this manually.

**Pull request closed without merging**